codelab-master/apps/playground/src/app/quiz/components/question/question.component.html

```diff
@@ -0,0 +1,82 @@
+<ng-content></ng-content>
+
+<form [formGroup]="formGroup">
+  <div class="options" *ngFor="let option of currentQuestion.options; index as i">
+    <div class="single-answer" *ngIf="!multipleAnswer">
+      <mat-radio-button
+        (change)="setSelected(i)"
+        [disabled]="alreadyAnswered"
+        [class.is-correct]="option.selected && option.correct"
+        [class.is-incorrect]="option.selected && !option.correct">
+
+        <mat-radio-label>
+          <span>{{ i + 1 }}. {{ option.text }}</span>
+
+          <mat-icon class="feedback-icon"
+                    *ngIf="option.selected && option.correct && isCorrectAnswerSelected">done</mat-icon>
+          <mat-icon class="feedback-icon"
+                    *ngIf="option.selected && !option.correct">clear</mat-icon>
+        </mat-radio-label>
+      </mat-radio-button>
+
+      <section class="messages">
+        <div *ngIf="option.selected && option.correct && isCorrectAnswerSelected">
+          <span>
+            <mat-icon class="sentiment-icon">sentiment_very_satisfied</mat-icon>&nbsp;&nbsp;&nbsp;
+            <span class="message correct-message" i18n>&nbsp;&nbsp;&nbsp;You&apos;re right! {{ correctMessage }}</span>
+          </span>
+        </div>
+        <div *ngIf="option.selected && !option.correct">
+          <span>
+            <mat-icon class="sentiment-icon">sentiment_very_dissatisfied</mat-icon>&nbsp;&nbsp;&nbsp;
+            <span class="message wrong-message" i18n>&nbsp;&nbsp;&nbsp;That&apos;s wrong. {{ correctMessage }}</span>
+          </span>
+        </div>
+      </section>
+      <!-- TODO: get this working, except with previousUserAnswersText -->
+      <!-- <div *ngFor="let previousOption of previousUserAnswers[i]" class="previous-answer-text">
+        <span *ngIf="option.selected && previousUserAnswers[i].length === 1">
+          Option {{ previousOption }} was selected in your previous quiz.
+        </span>
+      </div> -->
+    </div>
+
+    <div class="multiple-answer" *ngIf="multipleAnswer">
+      <mat-checkbox
+        (change)="setSelected(i)"
+        [class.is-correct]="option.selected && option.correct"
+        [class.is-incorrect]="option.selected && !option.correct">
```

Comment:
seems lime duplicate code? Can we move it out and share?

Reply:
yes, mat-radio-button and mat-checkbox have the same attributes (actually mat-radio also has a disabled property), not sure how to group these attributes as an object that can be included in both.

Reply:
I've added two string variables to the ts file for the correct and incorrect logic and replaced the logic in the template with those variables.

Reply:
As mentioned before, I tried moving the duplicate code out to the QuizService and replacing each condition with single variable (isCorrectOption/isIncorrectOption), but the class binding doesn't seem to work.

Reply:
As mentioned before, I tried moving the duplicate code out to the QuizService and replacing each condition with single variable (isCorrectOption/isIncorrectOption), but the class binding doesn't seem to work.

Reply:
I added className to my model and a couple more fields in the TS file, and now it works!

Reply:
I tried adding a QuestionService to eliminate the duplicate code in the TS files, but now the options are not clickable...